### PR TITLE
Repackage everything with node2nix

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

Please have a look at the commits to see a detailed description of the changes.
Further effort on #31032

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Ma27** [comment]
started `nox-review`, will leave a comment when this is done.
(/cc @svanderburg )

**Comment by grahamc** [comment]
Ok so how does this reduce the LOC by 60,000?

File: pkgs/tools/networking/airfield/default.nix [hunk 1]
```diff
@@ -1,39 +1,72 @@
...
   '';
 
-  passthru.names = ["Airfield"];
+  postFixup = ''
+    chmod +x $out/bin/airfield
+    wrapProgram $out/bin/airfield \
+      --set NODE_PATH "${runtimeEnv}/lib/node_modules"
+  '';
 
   meta = {
```
Comment (Ma27) [inline pkgs/tools/networking/airfield/default.nix]: meta might break

**Comment by Ma27** [comment]
@grahamc since I dropped the last remaining references of npm2nix this ancient package set could finally be removed.
I'm currently running `nox-review` against this PR locally to ensure that nothing breaks because of this.

**Comment by Ma27** [comment]
nox-review exited with code 0 on my machine.

**Comment by Ma27** [comment]
btw I resolved the merge conflicts...

**Comment by c0bw3b** [comment]
Great work here!

Bonus: this should close #24271

**Comment by copumpkin** [comment]
cc @svanderburg

**Comment by svanderburg** [comment]
Great work!

**Pull request merged**